tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -88,3 +98,104 @@ def test_write(self):
 
         self.start_consumer()
         self.run_validation()
+
+
+class ShadowIndexingWhileBusyTest(PreallocNodesTest):
+    # With SI enabled, run common operations against a cluster
+    # while the system is under load (busy).
+    # This class ports Test6 from https://github.com/redpanda-data/redpanda/issues/3572
+    # into ducktape.
+
+    segment_size = 20 * 2**20
+    topics = [TopicSpec()]
+
+    def __init__(self, test_context: TestContext):
+        si_settings = SISettings(log_segment_size=self.segment_size,
+                                 cloud_storage_cache_size=20 * 2**30,
+                                 cloud_storage_bucket='while-busy-bucket',
+                                 cloud_storage_enable_remote_read=False,
+                                 cloud_storage_enable_remote_write=False)
+
+        super(ShadowIndexingWhileBusyTest,
+              self).__init__(test_context=test_context,
+                             node_prealloc_count=1,
+                             num_brokers=7,
+                             si_settings=si_settings)
+
+    def setUp(self):
+        # Dedicated nodes refers to non-container nodes such as EC2 instances
+        self.topics[
+            0].partition_count = 100 if self.redpanda.dedicated_nodes else 10
+
+        # Topic creation happens here
+        super().setUp()
+
+        # Remote write/read and retention set at topic level
+        rpk = RpkTool(self.redpanda)
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.write', 'true')
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.read', 'true')
```

Comment:
is there any particular reason to enable it per topic? why no to set it to `true` in `__init__`?

Reply:
The original test was influenced by a customer issue where SI is enabled at the topic level. So this code is overflow from the original test.

Reply:
I think it's OK to enable SI at the topic level here so we have some overlap with user issues.